x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.ts

```diff
@@ -0,0 +1,323 @@
+
+    if (actionSendResponse.status === 'error') {
+      this.log.error(stringify(actionSendResponse));
+
+      // TODO TC: check if we want to handle different errors like 401
+      throw new ResponseActionsClientError(
+        `Attempt to send [${actionType}] to Crowdstrike failed: ${
+          actionSendResponse.serviceMessage || actionSendResponse.message
+        }`,
+        500,
+        actionSendResponse
+      );
+    }
+
+    this.log.debug(`Response:\n${stringify(actionSendResponse)}`);
+
+    return actionSendResponse;
+  }
+
+  private async getAgentDetails(
+    id: string
+  ): Promise<CrowdstrikeGetAgentsResponse['resources'][number]> {
+    const { id: connectorId } = await this.getConnector();
+    const executeOptions: CrowdstrikeConnectorExecuteOptions<CrowdstrikeGetAgentsParams> = {
+      actionId: connectorId,
+      params: {
+        subAction: SUB_ACTION.GET_AGENT_DETAILS,
+        subActionParams: {
+          ids: [id],
+        },
+      },
+    };
+
+    let crowdstrikeApiResponse: CrowdstrikeGetAgentsResponse | undefined;
+
+    try {
+      const response = (await this.connectorActionsClient.execute(
+        executeOptions
+      )) as ActionTypeExecutorResult<CrowdstrikeGetAgentsResponse>;
+
+      this.log.debug(`Response for Crowdstrike agent id [${id}] returned:\n${stringify(response)}`);
+
+      crowdstrikeApiResponse = response.data;
+    } catch (err) {
+      throw new ResponseActionsClientError(
+        `Error while attempting to retrieve Crowdstrike host with agent id [${id}]`,
+        500,
+        err
+      );
+    }
+
+    if (!crowdstrikeApiResponse || !crowdstrikeApiResponse.resources[0]) {
+      throw new ResponseActionsClientError(`Crowdstrike agent id [${id}] not found`, 404);
+    }
+
+    return crowdstrikeApiResponse.resources[0];
+  }
+
+  protected async validateRequest(
+    payload: ResponseActionsClientWriteActionRequestToEndpointIndexOptions
+  ): Promise<ResponseActionsClientValidateRequestResponse> {
+    // TODO:PT support multiple agents
+    if (payload.endpoint_ids.length > 1) {
+      return {
+        isValid: false,
+        error: new ResponseActionsClientError(
+          `[body.endpoint_ids]: Multiple agents IDs not currently supported for Crowdstrike`,
+          400
+        ),
+      };
+    }
+
+    return super.validateRequest(payload);
+  }
+
+  async isolate(
+    actionRequest: IsolationRouteRequestBody,
+    options: CommonResponseActionMethodOptions = {}
+  ): Promise<ActionDetails> {
+    const reqIndexOptions: ResponseActionsClientWriteActionRequestToEndpointIndexOptions = {
+      ...actionRequest,
+      ...this.getMethodOptions(options),
+      command: 'isolate',
+    };
+
+    if (!reqIndexOptions.error) {
+      let error = (await this.validateRequest(reqIndexOptions)).error;
+
+      if (!error) {
+        try {
+          await this.sendAction(SUB_ACTION.HOST_ACTIONS, {
+            ids: actionRequest.endpoint_ids,
```

Comment:
Question: does the S1 API accept any other parameters? like a comment? if so, then please populate that with something like this: `Taken from Elastic Security via Response Action ID {ationId}`. This will provide us with a reference in Crowdstrike back to Elastic.


Also - What is returned from this API call? Is there a "action id" from crowdstrike that we need to store along with our response action request? If so, my PR introduces a way to do that (store data with our Action Request and Action Response documents). This data will be needed to then check to see when the action is complete in Crowdstrike